### ISSUE TITLE
iTAK Compatibility

### ISSUE THREAD

**Issue by kavinsky** [kdudkov/goatak#8]
We can't connect using iTAK, using TCP nor TCP SSL.

Is iTAK supported?

**Comment by kdudkov** [comment]
yes, iTAK support was fixed some time ago. Do you use your own or community server?

**Comment by kavinsky** [comment]
Im using my own server, WinTAK and ATAK connects straight away but iTAK shows disconnected, we have tried building the datapackage in different forms as described in https://mytecknet.com/lets-build-a-tak-server/#iphone-tak-itak blog posts and using QR code but no joy.

**Comment by mki0331** [comment]
Hi,
nice project.

Same thing here, own server on a clean installed raspberry pi 4, test with iTAK v2.8.0(build 612)

SSL:
Log output:

{"level":"info","ts":1701935094.0372405,"caller":"goatak_server/tcpserver.go:64","msg":"SSL connection from 10.xxx.xx.31:50548"}
{"level":"error","ts":1701935094.1444004,"caller":"goatak_server/tcpserver.go:67","msg":"Handshake error: &errors.errorString{s:\"EOF\"}","stacktrace":"main.(*App).listenTls\n\tgithub.com/kdudkov/goatak/cmd/goatak_server/tcpserver.go:67\nmain.(*App).Run.func3\n\tgithub.com/kdudkov/goatak/cmd/goatak_server/main.go:145"}

TCP:
no log output, error msg from client: Userauth failed

**Comment by mki0331** [comment]
I think, there is a ssl prob:
in file cert_api.go(line 59):
names := map[string]string{"C": "RU", "O": "goatak", "OU": "goatak"}

and make_ca.sh differs from that.

**Comment by kdudkov** [comment]
> I think, there is a ssl prob: in file cert_api.go(line 59): names := map[string]string{"C": "RU", "O": "goatak", "OU": "goatak"}
> 
> and make_ca.sh differs from that.

Let me clarify a little: to work with ssl in atak you need two different certs for now:
- one for tcp stream (port 8999) and marti api (port 8443), this cert can be made by hands or via cert enrollment mechanism (afaik for itak only enrollment is an option)
- and one for cert enrollment (port 8446)

second cert must be verifiable by your device, so I prefere to use  letsencrypt and nginx proxy to deal with certs (recipe is [here](https://github.com/kdudkov/goatak/wiki/Server-Fast-Start#tls-setup)

May be you can use one self-signed cert for both, but you need to put ca cert to your device truststore.

I can't test with iTAK, because has no apple devices nearby, so I don't know if it is a way to use iTAK with tcp connection

**Comment by kavinsky** [comment]
Was problem on my end, ITAK requires cert enrollment in order to work. thank you.
